### ISSUE TITLE
Misleading error message when class used in `IClassFixture<>` throws exception in constructor

### ISSUE THREAD

**Issue by thomhurst** [xunit/xunit#2927]
Steps:
- Create a class that throws an exception in its constructor
- Use that class in an `IClassFixture<>`
- Run tests in the test class that defines the `IClassFixture<>`

Reproduction is here: https://github.com/thomhurst/xUnitFixtureDataErrorMessageReproduction

The error message implies that the data setup for the test is wrong, when in reality an exception was thrown.

```
Xunit.Sdk.TestClassException
The following constructor parameters did not have matching fixture data: MyData myData
```

**Comment by bradwilson** [comment]
It is showing the original exception as well:

![image](https://github.com/xunit/xunit/assets/16944/4529d3eb-a32e-4d20-ad01-16cb938c8605)

Updating the message to mention the reason why the constructor parameter couldn't match is an easy fix for me to make.

**Comment by bradwilson** [comment]
This had already been improved in v3 with a better message (`"Class fixture type 'TYPENAME' threw in its constructor"`), so I back-ported a bit of the logic to get this message now in v2 as well.

Available in v2 `2.8.1-pre.13` https://xunit.net/docs/using-ci-builds

Here's the output from your sample now:

![image](https://github.com/xunit/xunit/assets/16944/036b324a-58ef-4b53-83f8-9e385d2607cb)

**Comment by thomhurst** [comment]
Thanks @bradwilson !
